backend/compact-connect/lambdas/staff-user-pre-token/user_scopes.py

```diff
@@ -17,23 +19,30 @@ def _get_scopes_from_db(self, sub: str):
         :param sub: The `sub` field value from the Cognito Authorizer (which gets it from the JWT)
         """
         user_data = self._get_user_data(sub)
-        permissions = user_data.get('permissions', {})
+        permissions = {
+            compact_record['compact']: {
+                'actions': set(compact_record['permissions']['actions']),
+                'jurisdictions': compact_record['permissions']['jurisdictions']
+            }
+            for compact_record in user_data
+        }
 
         # Ensure included compacts are limited to supported values
-        disallowed_compcats = permissions.keys() - config.compacts
-        if disallowed_compcats:
-            raise ValueError(f'User permissions include disallowed compacts: {disallowed_compcats}')
+        disallowed_compacts = permissions.keys() - config.compacts
+        if disallowed_compacts:
+            raise ValueError(f'User permissions include disallowed compacts: {disallowed_compacts}')
```

Comment:
just to confirm, since this seems like a pretty critical check, we have a test case for this logic here right?

Reply:
Yep, [right here](https://github.com/InspiringApps/JCC-CompactConnect/blob/c1d2b92a0bde13a1367b416a03f2d3f268663e4e/backend/compact-connect/lambdas/staff-user-pre-token/tests/test_user_scopes.py#L140).